vmdk_plugin/main.go

```diff
@@ -97,17 +99,31 @@ func main() {
 	var driver volume.Driver
 
 	// Define command line options
-	driverName := flag.String("driver", "vmdk", "Volume driver")
-	logLevel := flag.String("log_level", "debug", "Logging Level")
+	logLevel := flag.String("log_level", "info", "Logging Level")
 	configFile := flag.String("config", config.DefaultConfigPath, "Configuration file path")
 
+	// Load the configuration if one was provided.
+	c, err := config.Load(*configFile)
+	if err != nil {
+		log.Warning("Failed to load config file %s: %v", *configFile, err)
+	}
+
+	// Default driver, overridden by the config file
+	name := vsphereDriver
+	if c.Driver != "" {
```

Comment:
Reordered the code to select the driver if no error in config file, logs below,

2017-02-27 20:54:44.414369896 -0800 PST [INFO] No config file found. Using defaults.
2017-02-27 20:54:44.415016096 -0800 PST [WARNING] Failed to load config file %s: %v/etc/docker-volume-vsphere.confopen /etc/docker-volume-vsphere.conf: no such file or directory
2017-02-27 20:54:44.415259471 -0800 PST [INFO] Starting plugin log_level=info config="/etc/docker-volume-vsphere.conf" driver=vsphere
2017-02-27 20:54:44.415492463 -0800 PST [INFO] Plugin options - port=1019
2017-02-27 20:54:44.415743608 -0800 PST [INFO] Getting volume data from unix:///var/run/docker.sock
2017-02-27 20:54:44.437056322 -0800 PST [INFO] Discovered 0 volumes in use.
2017-02-27 20:54:44.437362054 -0800 PST [INFO] Docker VMDK plugin started mock_esx=false version="vSphere Volume Driver v0.4" port=1019
2017-02-27 20:54:44.437833239 -0800 PST [INFO] Going into ServeUnix - Listening on Unix socket address="/run/docker/plugins/vsphere.sock"